notebooks/model_comp.qmd

```diff
@@ -0,0 +1,158 @@
+
+```{r}
+#| output: false
+library(tidyverse)
+library(tidybayes)
+library(fs)
+library(cmdstanr)
+library(posterior)
+library(jsonlite)
+library(scales)
+library(here)
+ci_width <- c(0.5, 0.8, 0.95)
+```
+
+## Load Data
+
+```{r}
+hosp_data <- tibble(.value = here(path("notebooks",
+  "data", "fit_hosp_only", "stan_data",
+  ext = "json"
+)) |>
+  jsonlite::read_json() |>
+  pluck("hosp") |>
+  unlist()) |>
+  mutate(time = row_number())
+
+stan_files <-
+  dir_ls(here(path("notebooks", "data", "fit_hosp_only")), glob = "*wwinference*") |>
+  enframe(name = NULL, value = "file_path") |>
+  mutate(file_details = path_ext_remove(path_file(file_path))) |>
+  separate_wider_delim(file_details,
+    delim = "-",
+    names = c("model", "date", "chain", "hash")
+  ) |>
+  mutate(date = ymd_hm(date)) |>
+  filter(date == max(date)) |>
+  pull(file_path)
+
+
+stan_tidy_draws <- read_cmdstan_csv(stan_files)$post_warmup_draws |> tidy_draws()
+
+pyrenew_tidy_draws <- read_csv(here(path("notebooks",
+  "data", "fit_hosp_only", "pyrenew_inference_data",
+  ext = "csv"
+))) |>
+  rename_with(\(varname) str_remove_all(varname, "\\(|\\)|\\'|(, \\d+)")) |>
+  rename(
+    .chain = chain,
+    .iteration = draw
+  ) |>
+  mutate(across(c(.chain, .iteration), \(x) as.integer(x + 1))) |>
+  mutate(
+    .draw = tidybayes:::draw_from_chain_and_iteration_(.chain, .iteration),
+    .after = .iteration
+  ) |>
+  pivot_longer(-starts_with("."),
+    names_sep = ", ",
+    names_to = c("distribution", "name")
+  ) |>
+  {
+    \(x) split(x |> select(-distribution), f = as.factor(x$distribution))
+  }() |>
+  map(\(x) pivot_wider(x, names_from = name) |> tidy_draws())
+```
+
+## Calculate Credible Intervals for Plotting
+
+```{r}
+combined_ci_for_plotting <-
+  bind_rows(
+    pyrenew_tidy_draws$posterior_predictive |>
+      gather_draws(observed_hospital_admissions[time], rt[time], ihr[time]) |>
+      median_qi(.width = ci_width) |>
+      mutate(model = "pyrenew"),
+    stan_tidy_draws |>
+      gather_draws(pred_hosp[time], rt[time], p_hosp[time]) |>
+      mutate(.variable = case_when(
+        .variable == "pred_hosp" ~ "observed_hospital_admissions",
+        .variable == "p_hosp" ~ "ihr",
+        TRUE ~ .variable
+      )) |>
+      median_qi(.width = ci_width) |>
+      mutate(model = "stan")
+  )
+```
+
+
+
+## Hospital Admission Comparison
+
+```{r}
+combined_ci_for_plotting |>
+  filter(.variable == "observed_hospital_admissions") |>
+  ggplot(aes(time, .value)) +
+  facet_wrap(~model) +
+  geom_lineribbon(aes(ymin = .lower, ymax = .upper), color = "#08519c") +
+  scale_fill_brewer(
+    name = "Credible Interval Width",
+    labels = ~ percent(as.numeric(.))
+  ) +
+  geom_point(data = hosp_data) +
+  cowplot::theme_cowplot() +
+  ggtitle("Vignette Data Model Comparison") +
+  scale_y_continuous("Hospital Admissions") +
+  scale_x_continuous("Time") +
+  theme(legend.position = "bottom")
+```
+
+
+
+## Rt Comparions
+
```

Comment:
Maybe add a note here to explain why the two Rt estimates are different. Due to the lower prior variance for Rt in pyrenew model, right?

Reply:
The Rt comparisons should not be different. You are probably seeing results from a previous Stan fit and may need to rerun the Stan model.

Reply:
Perhaps, but I'm not completely sure. They seem to basically be in agreement after `t=0`
![unnamed-chunk-5-1](https://github.com/user-attachments/assets/9ce907ce-53b4-4992-82d6-bc71d97ddcc6)